src/features/device-proxy/device-proxy.ts

```diff
@@ -184,29 +184,47 @@ export async function requestDevices({
 	if (!['PUT', 'PATCH', 'POST', 'HEAD', 'DELETE', 'GET'].includes(method)) {
 		throw new BadRequestError(`Invalid method '${method}'`);
 	}
-	const deviceIds = await api.resin.get({
-		resource: 'device',
-		options: {
-			$select: 'id',
-			$filter: {
-				$and: [
-					{
-						is_connected_to_vpn: true,
-						vpn_address: { $ne: null },
-					},
-					filter,
-				],
-			},
-		},
+	const resinApi = api.resin.clone({
 		passthrough: { req },
 	});
+	const deviceIds = (
+		await resinApi.get({
+			resource: 'device',
+			options: {
+				$select: 'id',
+				$filter: {
+					$and: [
+						{
+							is_connected_to_vpn: true,
+							vpn_address: { $ne: null },
+						},
+						filter,
+					],
+				},
+			},
+		})
+	).map(({ id }) => id) as number[];
```

Comment:
I'd prefer this as
```suggestion
	) as Device[]).map(({ id }) => id);
```
or
```suggestion
	) as Pick<Device, 'id'>[]).map(({ id }) => id);
```
so that we type the response closer to the source and therefore benefit from the improved typings for more places (eg after the change typescript would be able to catch a typo like `.map(({ iid }) => iid)` whereas before that would be a runtime error instead)

Reply:
Makes sense :+1: